posts/src/main/kotlin/pl/edu/pw/ia/posts/domain/query/projector/PostProjector.kt

```diff
@@ -0,0 +1,70 @@
+package pl.edu.pw.ia.posts.domain.query.projector
+
+import org.axonframework.eventhandling.EventHandler
+import org.axonframework.queryhandling.QueryGateway
+import org.axonframework.queryhandling.QueryHandler
+import org.springframework.data.domain.Page
+import org.springframework.stereotype.Service
+import pl.edu.pw.ia.posts.domain.query.repository.PostViewRepository
+import pl.edu.pw.ia.shared.domain.event.PostCreatedEvent
+import pl.edu.pw.ia.shared.domain.event.PostDeletedEvent
+import pl.edu.pw.ia.shared.domain.event.PostUpdatedEvent
+import pl.edu.pw.ia.shared.domain.exception.PostNotFoundException
+import pl.edu.pw.ia.shared.domain.query.FindAccountByIdQuery
+import pl.edu.pw.ia.shared.domain.query.FindPostByIdQuery
+import pl.edu.pw.ia.shared.domain.query.FindPostsByContentAndThreadIdQuery
+import pl.edu.pw.ia.shared.domain.query.FindPostsByContentQuery
+import pl.edu.pw.ia.shared.domain.query.FindPostsByThreadIdQuery
+import pl.edu.pw.ia.shared.domain.view.AccountView
+import pl.edu.pw.ia.shared.domain.view.PostView
+
+@Service
+class PostProjector(
+	private val repository: PostViewRepository,
+	private val queryGateway: QueryGateway
+) {
+	@EventHandler
+	fun on(event: PostCreatedEvent) {
+		val username = queryGateway.query(
+			FindAccountByIdQuery(accountId = event.accountId),
+			AccountView::class.java
+		).get().name
+		val view = PostView(
+			postId = event.postId,
+			threadId = event.threadId,
+			accountId = event.accountId,
+			username = username,
+			content = event.content,
+			createdAt = event.createdAt,
+			votes = 0
+		)
+		repository.save(view)
+	}
+
+	@EventHandler
+	fun on(event: PostUpdatedEvent) {
+		val view = repository.findById(event.postId) ?: throw PostNotFoundException(event.postId)
+		repository.save(view.copy(content = event.content))
+	}
+
```

Comment:
here we need a `on(event: AccountUpdatedEvent)`
In case a user modifies their username

Reply:
and a `on(event: VoteCreatedEvent)`, which will increase the number of votes
and a `on(event: VoteUpdatedEvent)` which will update the number of votes based on the previous vote value

Reply:
and a `on(event: VoteCreatedEvent)`, which will increase the number of votes
and a `on(event: VoteUpdatedEvent)` which will update the number of votes based on the previous vote value